### PR TITLE
client side Validation

### LINKED ISSUE #46: validate email before sending
We can just use the FF built-in validation. It's lenient (allows `user@localhost`) but RFC compliant.

https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Constraint_validation

### PR DESCRIPTION
**Pull request by meandavejustice**
* Handles showing the user messages based on server response
* client side email validation.
* gulp task for building js
* sourcemaps

fixes #46 #38 #37 #69

### CODE REVIEW

**Comment by meandavejustice** [comment]
cc @johngruen

**Comment by johngruen** [comment]
@meandavejustice looks good. I am at dinner, so if this is blocking graybeard bet someone else to review formally and merge. Otherwise will do when I get home.

**Comment by johngruen** [comment]
Autocorrect it truly strange

**Comment by meandavejustice** [comment]
@johngruen not blocking anything, enjoy your dinner. This can be review whenever.

**Comment by 6a68** [comment]
I'm out of time and will be gone till Tuesday. Les, mind reviewing this'un?

**Comment by johngruen** [comment]
hey @meandavejustice This looks good, but I'd like to change things around to start with a disabled form and do validation checks on `keyup` events in the email field. 

I want to spend some time polishing this since it's like the only interaction on the page. Do you mind if I make a PR into your branch?

**Comment by meandavejustice** [comment]
@johngruen go for it

**Comment by lmorchard** [comment]
I'm getting my ideatown on today (finally), so I should be good to review once you guys have some more changes settled in

**Comment by johngruen** [comment]
@lmorchard want to review this when you get a chance?

File: src/scripts/main.js [hunk 1]
```diff
@@ -1,14 +1,103 @@
...
+
+var SLOW = 1000;
+var FAST = 150;
+
+emailEl.addEventListener('keyup', function(ev) {
+
+  var emailVal = emailEl.value;
+  var isValid = validateEmail(emailVal);
+
+  if(ev.charCode !== 13) {
```
Comment (lmorchard) [inline src/scripts/main.js]: I think you want `keyCode !== 13` here. Right now, the message is kind of popping in and disappearing repeatedly when I hit enter.

File: gulpfile.js [hunk 2]
```diff
@@ -12,12 +11,17 @@ var gulp = require('gulp'),
...
+  gutil = require('gulp-util'),
   uglify = require('gulp-uglify'),
-  vfs = require('vinyl-fs');
+  sourcemaps = require('gulp-sourcemaps'),
+  vfs = require('vinyl-fs'),
+  browserify = require('browserify'),
+  source = require('vinyl-source-stream'),
+  buffer = require('vinyl-buffer'),
+  del = require('del'),
+  runSequence = require('run-sequence'),
```
Comment (lmorchard) [inline gulpfile.js]: I don't think `vfs`, `del`, or `runSequence` get used anymore. Might be worth removing.

**Comment by lmorchard** [comment]
Other than my idle IRC grumbling, looks good to me!

**Pull request merged**